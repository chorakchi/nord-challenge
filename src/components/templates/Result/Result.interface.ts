import { HTMLAttributes } from "react";

type HeaderItem = {
  key: string;
  label: string;
};

export interface TypographyProps extends HTMLAttributes<HTMLElement> {
  logout?: () => void;
  header?: Array<HeaderItem>;
  onClickHeader?: (key: string) => void;
  items?: Array<any>;
}
