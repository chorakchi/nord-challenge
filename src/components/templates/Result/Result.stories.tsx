import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Result } from "./Result";
import { mockData } from "./Result.mock";

export default {
  title: "templates/Result",
  component: Result,
  parameters: {
    docs: {
      description: {
        component: "",
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof Result>;

const Template: ComponentStory<typeof Result> = (args) => <Result {...args} />;

export const Result_ = Template.bind({});
Result_.args = { ...mockData };
