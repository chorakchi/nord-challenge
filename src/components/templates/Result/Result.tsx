import { TypographyProps } from "./Result.interface";
import { Button, Container, ResultBlock } from "components";
export const Result: React.FC<TypographyProps> = ({
  logout,
  ...props
}: TypographyProps) => {
  return (
    <Container>
      <ResultBlock {...props}></ResultBlock>
      <div className="w-32 absolute top-10 right-10">
        <Button onClick={logout}>logout</Button>
      </div>
    </Container>
  );
};
