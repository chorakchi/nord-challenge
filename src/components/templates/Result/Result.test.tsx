import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Result } from "./Result";

describe("Result", () => {
  it("should render the Result", () => {
    const subject = shallow(<Result/>);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer
      .create(<Result/>)
      .toJSON();
    expect(subject).toMatchSnapshot();
  });
});
