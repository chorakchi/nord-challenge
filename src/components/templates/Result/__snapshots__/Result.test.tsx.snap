// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render correctly and match to the Snapshot 1`] = `
<div
  className="flex flex-col h-screen container mx-auto justify-center"
>
  <div
    className="fade-in min-h-full flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8"
  >
    <div
      className="w-32 absolute top-10 right-10"
    >
      <button
        className="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      >
        logout
      </button>
    </div>
  </div>
</div>
`;
